src/databricks/labs/ucx/source_code/queries.py

```diff
@@ -1,53 +1,107 @@
+from dataclasses import asdict, dataclass
 from datetime import datetime
 
 from databricks.sdk import WorkspaceClient
 from databricks.sdk.service.sql import ListQueryObjectsResponseQuery
+from databricks.sdk.service.workspace import Language
 
+from databricks.labs.lsql.backends import SqlBackend
 
-from databricks.labs.ucx.source_code.directfs_access import DirectFsAccessCrawler, DirectFsAccessInQuery
+from databricks.labs.ucx.hive_metastore.table_migration_status import TableMigrationIndex
+from databricks.labs.ucx.source_code.base import CurrentSessionState
+from databricks.labs.ucx.source_code.directfs_access import DirectFsAccessCrawler, DirectFsAccessInQuery, LineageAtom
+from databricks.labs.ucx.source_code.linters.context import LinterContext
 from databricks.labs.ucx.source_code.linters.directfs import DirectFsAccessSqlLinter
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class QueryProblem:
+    query_parent: str
+    query_name: str
+    code: str
+    message: str
+
+
 class QueryLinter:
 
     def __init__(
         self,
         ws: WorkspaceClient,
+        migration_index: TableMigrationIndex,
         directfs_crawler: DirectFsAccessCrawler,
     ):
         self._ws = ws
+        self._migration_index = migration_index
         self._directfs_crawler = directfs_crawler
 
-    def collect_dfsas_from_queries(self) -> Iterable[DirectFsAccessInQuery]:
+    def refresh_report(self, sql_backend: SqlBackend, inventory_database: str):
         assessment_start = datetime.now()
-        queries_dfsas = self._collect_dfsas_from_queries()
-        assessment_end = datetime.now()
-        dfsas: list[DirectFsAccessInQuery] = []
-        for dfsa in queries_dfsas:
-            dfsas.append(
-                dfsa.replace_assessment_infos(assessment_start=assessment_start, assessment_end=assessment_end)
-            )
-        self._directfs_crawler.dump_all(dfsas)
-        return dfsas
+        all_queries = list(self._ws.queries.list())
+        logger.info(f"Running {len(all_queries)} linting tasks...")
+        query_problems: list[QueryProblem] = []
+        query_dfsas: list[DirectFsAccessInQuery] = []
+        for query in all_queries:
+            problems = self.lint_query(query)
+            query_problems.extend(problems)
+            dfsas = self.collect_dfsas_from_query(query)
+            assessment_end = datetime.now()
+            for dfsa in dfsas:
+                query_dfsas.append(
+                    dataclasses.replace(
+                        dfsa, assessment_start_timestamp=assessment_start, assessment_end_timestamp=assessment_end
+                    )
+                )
+        logger.info(f"Saving {len(query_problems)} linting problems...")
+        sql_backend.save_table(
+            f'{inventory_database}.query_problems',
+            query_problems,
+            QueryProblem,
+            mode='overwrite',
+        )
+        self._directfs_crawler.dump_all(query_dfsas)
```

Comment:
This one surprised me as it is different to the useful flow: tables updated from one location. I see a TODO in the "dump all" about changing it when we support history. I have to think about the implications, just noting ti for now